docs/snapshot-controller.md

```diff
@@ -0,0 +1,132 @@
+flowchart TD
+  %% Defining the styles
+    classDef Red fill:#FF9999;
+    classDef Amber fill:#FFDEAD;
+    classDef Green fill:#BDFFA4;
+
+  predicate((PREDICATE: <br>Snapshot got created OR <br> changed to Finished))
+
+  %%%%%%%%%%%%%%%%%%%%%%% Drawing EnsureAllIntegrationTestPipelinesExist() function 
+
+  %% Node definitions
+  ensure1(Process further if: Snapshot testing <br>is not finished yet)
+  are_there_any_ITS{"Are there any <br>IntegrationTestScenario <br>present for the given <br>Application?"}
+  does_ITS_has_env_defined{Does the <br>IntegrationTestScenario <br>has any environment <br>defined in it?}
+  skip_creating_test_PLR(Skip creating Test PLR for this ITS,<br> as it will be created by binding controller)
+  create_new_test_PLR(<b>Create a new Test PipelineRun</b> for each <br>of the above ITS, if it doesn't exists already)
+  mark_snapshot_InProgress(<b>Mark</b> Snapshot's Integration-testing <br>status as 'InProgress')
+  fetch_all_required_ITS("Fetch all the required <br>(non-optional) IntegrationTestScenario <br>for the given Application")
+  encountered_error1{Encountered error?}
+  mark_snapshot_Invalid1(<b>Mark</b> the Snapshot as Invalid)
+  is_atleast_1_required_ITS{Is there atleast <br>1 required ITS?}
+  mark_snapshot_passed(<b>Mark</b> the Snapshot as Passed)
+  continue_processing1(Controller continues processing...)
+
+  %% Node connections
+  predicate                 ---->    |"EnsureAllIntegrationTestPipelinesExist()"|ensure1
+  ensure1                   -->      are_there_any_ITS
+  are_there_any_ITS         --Yes--> does_ITS_has_env_defined
+  are_there_any_ITS         --No-->  fetch_all_required_ITS
+  does_ITS_has_env_defined  --Yes--> skip_creating_test_PLR
+  does_ITS_has_env_defined  --No-->  create_new_test_PLR
+  skip_creating_test_PLR    -->      fetch_all_required_ITS
+  create_new_test_PLR       -->      mark_snapshot_InProgress
+  mark_snapshot_InProgress  -->      fetch_all_required_ITS
+  fetch_all_required_ITS    -->      encountered_error1
+  encountered_error1        --No-->  is_atleast_1_required_ITS
+  encountered_error1        --Yes--> mark_snapshot_Invalid1
+  is_atleast_1_required_ITS --Yes--> continue_processing1
+  is_atleast_1_required_ITS --No-->  mark_snapshot_passed
+  mark_snapshot_passed      -->      continue_processing1
+
+
+  %%%%%%%%%%%%%%%%%%%%%%% Drawing EnsureGlobalCandidateImageUpdated() function 
+
+  %% Node definitions
+  ensure2(Process further if: Component is not nil & <br>Snapshot testing succeeded & <br>Snapshot was not created by <br>PAC Pull Request Event)
+  update_container_image("<b>Update</b> the '.spec.containerImage' field of the given <br>component with the latest value, taken from <br>given Snapshot's .spec.components[x].containerImage field")
+  update_last_built_commit("<b>Update</b> the '.status.lastBuiltCommit' field of the given <br>component with the latest value, taken from <br>given Snapshot's .spec.components[x].source.git.revision field")
```

Comment:
I'm not sure if I understood your suggestion.
* Are you suggesting linking all the `"Controller continues processing..."` blocks to one another because that's how they work in reality? then that's accurate yes, but it would make the diagram a little messy and very zig-zag, and hence hard to read...wdyt? 

* Are you suggesting adding a new `"Controller continues processing..."` block at the end of the `EnsureGlobalCandidateImageUpdated()` branch, like this?

```mermaid
%%{init: {'theme':'forest'}}%%
flowchart TD

  %% Node definitions
  update_last_built_commit("<b>Update</b> the '.status.lastBuiltCommit' field of the given <br>component with the latest value, taken from <br>given Snapshot's .spec.components[x].source.git.revision field")
continue_processing("Controller continues processing...")

 %% Node connections
  update_last_built_commit -->   continue_processing

```

Reply:
Yes

Reply:
Updated, thanks very much for your time!!